shell/platform/fuchsia/dart-pkg/zircon_ffi/dart_dl.cc

```diff
@@ -0,0 +1,17 @@
+#include "dart_dl.h"
+
+#include "flutter/fml/logging.h"
+#include "include/dart_api_dl.h"
+
+int zircon_dart_dl_initialize(void* initialize_api_dl_data) {
+  if (Dart_InitializeApiDL(initialize_api_dl_data) != 0) {
```

Comment:
How are we gonna expose this through the embedder C API?  The version of dart used and access to the dart VM is hidden behind the embedder C API....

It works for now with the code in-engine-tree, but how will we guarantee that it will continue to work when we are consuming only embedder.h & libflutter.so?  What happens if the dart API versions of the embedder and engine diverge?  Does the Dart API also present a stable ABI?

Reply:
If the engine and embedding use diverging versions of dart we will also have AOT considerations to worry about. I think in the short-medium term we will be using the same version for both fuchsia-embedding and the engine. When we decide to allow divergence of dart versions, we can likely piggyback on the same solution as AOT to resolve this.

Also `Dart_InitializeApiDL` is currently only checking for a min version of dart. I think the `dart_api_dl.h` does give us a stable ABI, but I will confirm can report back to you.